tetris.py

```diff
@@ -117,59 +116,66 @@ def new_block(self):
         """
         x = int(game_width / 2)
         y = 0
-        blocks = {
-            "L": [(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)],
-            "L_rev": [(x - 1, y + 1), (x - 1, y), (x, y), (x + 1, y)],
-            "O": [(x - 1, y), (x, y), (x - 1, y + 1), (x, y + 1)],
-            "E": [(x, y), (x - 1, y + 1), (x, y + 1), (x + 1, y + 1)],
-            "Z": [(x - 1, y), (x, y), (x, y + 1), (x - 1, y + 1)],
-            "Z_rev": [(x + 1, y), (x, y), (x, y + 1), (x - 1, y + 1)],
-            "I": [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)],
-        }
-        if self.upcoming_block is None:
-            self.current_block = random.choice(list(blocks.values()))
+        if self.upcoming_block_shape is None:
+            self.current_block_shape = random.choice(shapes_name)
+            self.current_block_center = (x, y)
         else:
-            self.current_block = self.upcoming_block
-        self.upcoming_block = random.choice(list(blocks.values()))
+            self.current_block_shape = self.upcoming_block_shape
+            self.current_block_center = (x, y)
```

Comment:
The center is set in both cases, so it can be moved outside the `if`,`else`. In other words, do this:

```python3
if blah blah:
    self.current_block_shape = ...
else:
    self.current_block_shape = ...
self.current_block_center = (x, y)
```

Can't do this as a suggestion for whatever reason